compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -4456,6 +4461,9 @@ private void visitCheckAndCheckPanicExpr(BLangCheckedExpr checkedExpr) {
         String operatorType = checkedExpr.getKind() == NodeKind.CHECK_EXPR ? "check" : "checkpanic";
         boolean firstVisit = checkedExpr.expr.type == null;
         BType exprExpType;
+        if (checkedExpr.expr instanceof BLangFieldBasedAccess && checkedExpr.getKind() == NodeKind.CHECK_EXPR) {
```

Comment:
Instead of `instanceof` check, can we use `checkedExpr.expr.getKind() == NodeKind.FIELD_BASED_ACCESS_EXPR` ?

Reply:
Changed.